Block/BlockContextManager.php

```diff
@@ -37,6 +40,14 @@ class BlockContextManager implements BlockContextManagerInterface
     protected $logger;
 
     /**
+     * Used for deprecation check on {@link resolve} method.
+     * To be removed in 3.0 with BC system.
+     *
+     * @var array
+     */
+    private $reflectionCache;
```

Comment:
There is no need to cache reflection objects. See this benchmark: https://gist.github.com/mindplay-dk/3359812